compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs

```diff
@@ -107,6 +116,15 @@ data ShakeTestEnv = ShakeTestEnv
     , steVirtualResourcesNotes :: TVar (Map VirtualResource T.Text)
     }
 
+data ExpectedChoices = ExpectedChoices
+    { _cName :: String
+    , _consuming :: Bool
+    } deriving (Eq, Ord, Show )
+data TemplateProp = TemplateProp
+    { _choices :: Set.Set ExpectedChoices
+    , _action :: Int
```

Comment:
A comment explaining what the `Int` represents would be nice. It looks like it is the sum of the number of actions for each choice? In that case `numActions` might be a better field name. Seems like a slightly weird property to test for but we can always change or extend this later so I’m happy to keep it for now.

Reply:
Yes this `Int` is not adding any value, this is intended to check the action from a choice. I will fill this out in a following PR